internal/scheduler/task.go

```diff
@@ -0,0 +1,154 @@
+package scheduler
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/denisushakov/todo-rest/internal/storage/sqlite"
+	"github.com/denisushakov/todo-rest/pkg/models"
+)
+
+type Scheduler struct {
+	Storage *sqlite.Storage
+}
+
+func NewScheduler(dataBase *sqlite.Storage) *Scheduler {
+	return &Scheduler{
+		Storage: dataBase,
+	}
+}
+
+type TaskScheduler interface {
+	SaveTask(*models.Task) (int64, error)
+	GetTasks(string) ([]*models.Task, error)
+	GetTaskByID(string) (*models.Task, error)
+	UpdateTask(*models.Task) error
+	MarkTaskCompleted(string) error
+	DeleteTask(string) error
+}
+
+func (s *Scheduler) SaveTask(task *models.Task) (int64, error) {
+	if err := check(task); err != nil {
+		return 0, err
```

Comment:
Лучше оборачивать ошибки
```suggestion
		return 0, fmt.Errorf("erorr while saving task: %w", err)
```